core/types/block_compatibility_test.go

```diff
@@ -0,0 +1,44 @@
+package types
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestEthereumHeaderHashCompatibility(t *testing.T) {
```

Comment:
Should we add a similar test with a recent celo block to make it obvious that the hashing is stable there as well?

Reply:
Yes, that's easy to do and covers another case, so I'll do it!